### ISSUE TITLE
Wide gitsigns line

### ISSUE THREAD

**Issue by kobzar** [marko-cerovac/material.nvim#141]
A long time ago the material is my favorite color scheme. I am use it for all of i can. 
For editors, GUI, etc.
So, i am used this theme for my vim/neovim/spacevim/lunarvim as default.
But few time ago the theme is working wrong, i was write it in 
https://github.com/marko-cerovac/material.nvim/issues/120

For this time i found next 
All my settings is default. 
For fix my issue i add in plugin config gitsigns
```
	-- Material color scheme
	{
		"marko-cerovac/material.nvim",
		config = function()
			require("material").setup({
				plugins = {
					-- Available plugins:
					"dap",
					"dashboard",
					"gitsigns",
					"indent-blankline",
					"nvim-cmp",
					"nvim-tree",
					"telescope",
					"trouble",
					"which-key",
				},
			})
		end,
	},

```
But this doesn't resolve the problem. 
But! If after lvim is started if i run
```
:colorscheme material
```
The gitsigns line look normaly.
But after this command another plugin like colorizer goin to down. 
![image](https://user-images.githubusercontent.com/1912842/198829899-40c2810f-dd76-4b06-bc4d-10b4347ed64a.png)
 
Any other popular color themes like catpuccine,onedark, tokyonight e.t.c. working normaly without any parameters or additional configuration. All of i need just add the plugin name and choose the theme name. 

And as i say, a long time all working fine. 
The next question - why the plugin list is deactivated by default? If i choose the material theme, i want that all my enviroment  get this theme, not only for gitsigns.

**Comment by marko-cerovac** [comment]
I'll look into this issue over the weekend. It seems  that async loading is the source od the problem.

And the answer to your question is:
Material.nvim supports a lot of plugins. And highlights need to be applied for every plugin. But most users don't use every plugin that the theme supports. For example, some users only use telescope.nvim and gitsigns. This way, the theme only applies highlights for telescope and gitsigns. This reduces the startup time and allows for adding support for many plugins without slowing down the theme for users that don't use them.

**Comment by marko-cerovac** [comment]
@kobzar What is the output of: ```:highlight GitSignsAdd``` ?

**Comment by kobzar** [comment]
```
:highlight GitSignsAdd
GitSignsAdd    xxx links to DiffAdd
Press ENTER or type command to continue
```

**Comment by marko-cerovac** [comment]
I believe that this is an issue with your gitsigns config or a race condition with material.nvim and gitsigns.nvim.
Can you please show me your gitsigns config.

I have this in my config:
```lua
require("gitsigns").setup {
    signs = {
        add = { hl = "GitSignsAdd", text = "│", numhl = "GitSignsAddNr", linehl = "GitSignsAddLn" },
        change = { hl = "GitSignsChange", text = "│", numhl = "GitSignsChangeNr", linehl = "GitSignsChangeLn" },
        delete = { hl = "GitSignsDelete", text = "_", numhl = "GitSignsDeleteNr", linehl = "GitSignsDeleteLn" },
        topdelete = { hl = "GitSignsDelete", text = "‾", numhl = "GitSignsDeleteNr", linehl = "GitSignsDeleteLn" },
        changedelete = { hl = "GitSignsChange", text = "│", numhl = "GitSignsChangeNr", linehl = "GitSignsChangeLn" },
    },
}
```

This is the default I believe. This way, gitsigns doesn't link to ```DiffAdd```.

**Comment by kobzar** [comment]
https://github.com/LunarVim/LunarVim/blob/master/lua/lvim/core/gitsigns.lua

**Comment by eikenb** [comment]
> This is probably a compatibility issue with lunar-vim.

Just FYI it isn't lunar-vim. I have the same issue with no lunar-vim installed.

I do have the same thing where GitSignsAdd links to DiffAdd.

**Comment by kobzar** [comment]
Today i was update the material plugin and all working fine! 
and i dont change anything in my config. Just add plugin without any settings. 
Thanks!

**Comment by eikenb** [comment]
Still isn't working for me with the vim 0.8 and the latest material and gitsigns.

**Comment by kobzar** [comment]
Did you try to clean caches like
PackerClean PackerSync PackerCompile?

**Comment by eikenb** [comment]
Just ran PackerClean then re-ran PackerSync and PackerCompile (previously I'd only run the last 2). But it didn't help.

**Comment by marko-cerovac** [comment]
Gitsigns.nvim highlights work in a way where they use the ```GitSignsAdd``` group for added lines. If that group isn't already set, then ```GitGutterAdd```  gets set. If that group also isn't set, ```diffAdded``` gets used, then ```DiffAdd```. This creates a race condition. If gitsigns get the ```GitSignsAdd``` highlight group, everything looks fine, but if this doesn't get loaded in time, then it uses ```diffAdded``` which is reversed by default, causing the wide look. @eikenb I assume that in your config, material.nvim gets loaded after gitsigns.nvim. Can you please try and reverse the order and let me know if there's any changes.

**Comment by ggalindezb** [comment]
@marko-cerovac I had the same problem but after loading git signs after material, the groups load correctly as you suggested. As an extra bit of info, I do not have a full Lua config, I load Material with `colorscheme material` but do load GitSigns within a Lua block.

Thanks for the awesome colorscheme, btw.

**Comment by eikenb** [comment]
Thanks for the tips @marko-cerovac, sorry for delayed response.

I stripped out everything except gitsigns and material and am still having the issue. The only additional thing I included was the packer module (as I use packer). I'll just cut-n-paste my config below as it is very short now.

If I switch the order of the material and gitsigns `use` blocks in lua/plugins.vim and it makes no difference.

I test it by loading up vim with a random git controlled file and manually setting the colorscheme after the fact with `:colorscheme material`.

My init.lua file contains.
```lua
require("plugins")
```

My lua/plugins.lua contains.
```lua
return require("packer").startup(function(use)
  use "wbthomason/packer.nvim"

  use {
    'lewis6991/gitsigns.nvim',
    config = function()
      require('gitsigns').setup {}
    end
  }

  use {
    'marko-cerovac/material.nvim',
    config = function()
        require('material').setup{}
    end,
  }

 end)
```

Thanks!

**Comment by marko-cerovac** [comment]
@eikenb sorry for the **very** late response. I wasn't active lately and I didn't recieve any notifications from github or email about this issue for some reason.
I think that I have managed to pinpoint the source of the race condition.
You see, I've reversed the foreground and background colors for the DiffAdd, DiffDelete and DiffText to make the diff more visible like in [this](https://github.com/marko-cerovac/material.nvim/issues/141#issuecomment-1314517675) screenchot earlier.
And gitsigns.nvim highlights the signs by first checking if the user set them like I did here [here](https://github.com/marko-cerovac/material.nvim/issues/141#issuecomment-1303890098).
If he didn't, it then tries to inherit the color from some other highlight group and it checks for GitGutter, GitSignify, and so on...
In the end, It picks up the color from DiffAdd, DiffDelete and DiffText which is reversed.
Another thing you could try in your config is this:
```lua
require("material").setup {
    custom_highlights = {
        DiffAdd = { reverse = false },
        DiffChange = { reverse = false },
    }
}
```
I believe that the two possible solutions for this problem would be:
1. Define gitsigns in the setup function for the plugin (like in the code snippet that I have linked)
2. Do this in the custom_highlights table in the material setup function

Let me know if this this still doesn't work so that I can look for another solution.

**Comment by eikenb** [comment]
Hey @marko-cerovac, thanks for that.

I have it working now, but not exactly as you said. I added custom_highlights block to my setup and that stopped it from filling the background of the sign column with the color, but the fg color it uses the neutral editor.fg color (#b0bec5). It uses that same editor.fg color for each of the custom_highlights 'reverse=false` added.

To get colors I changed it to (also added DiffDelete as it had the same issue)...
```lua
c = require "material.colors.conditionals"
require("material").setup {
    custom_highlights = {
        DiffAdd = { fg = c.main.green, reverse = false },
        DiffChange = { fg = c.main.blue, reverse = false },
        DiffDelete = { fg = c.main.red, reverse = false },
    },
}
```
